cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -1234,3 +1254,10 @@ pProtocolVersion =
             <> Opt.help "Minor protocol version. An increase indicates a soft fork\
                         \ (old software canvalidate but not produce new blocks)."
             )
+
+--------------------------------------------------------------------------------
+-- Helpers
+--------------------------------------------------------------------------------
+
+fromDouble :: Double -> Maybe UnitInterval
+fromDouble = mkUnitInterval . toRational
```

Comment:
This isn't what you want. If `mkUnitInterval` returns nothing it's because the double value is outside of the 0..1 range and that should be a flag value parser error.

So we should turn this into a flag value parser for `UnitInterval` which does the error check.